api/v1alpha1/obtenantbackuppolicy_types.go

```diff
@@ -0,0 +1,127 @@
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.status`
+//+kubebuilder:printcolumn:name="TenantName",type=string,JSONPath=`.spec.tenantName`
+//+kubebuilder:printcolumn:name="NextFull",type=string,JSONPath=`.status.nextFull`
+//+kubebuilder:printcolumn:name="NextIncremental",type=string,JSONPath=`.status.nextIncremental`
+//+kubebuilder:printcolumn:name="FullCrontab",type=string,JSONPath=`.spec.dataBackup.fullCrontab`
+//+kubebuilder:printcolumn:name="IncrementalCrontab",type=string,JSONPath=`.spec.dataBackup.incrementalCrontab`
+
+// OBTenantBackupPolicy is the Schema for the obtenantbackuppolicies API
+type OBTenantBackupPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OBTenantBackupPolicySpec   `json:"spec,omitempty"`
+	Status OBTenantBackupPolicyStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// OBTenantBackupPolicyList contains a list of OBTenantBackupPolicy
+type OBTenantBackupPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OBTenantBackupPolicy `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&OBTenantBackupPolicy{}, &OBTenantBackupPolicyList{})
+}
+
+// LogArchiveConfig contains the configuration for log archive progress
+type LogArchiveConfig struct {
+	Destination         BackupDestination `json:"destination"`
+	SwitchPieceInterval string            `json:"switchPieceInterval"`
+	Binding             ArchiveBinding    `json:"binding,omitempty"`
+	DestDisabled        bool              `json:"destDisabled,omitempty"`
+	Concurrency         int               `json:"concurrency,omitempty"`
+}
+
+type ArchiveBinding string
+
+const (
+	ArchiveBindingOptional  = "Optional"
+	ArchiveBindingMandatory = "Mandatory"
+)
+
+// DataBackupConfig contains the configuration for data backup progress
+type DataBackupConfig struct {
+	Destination        BackupDestination `json:"destination"`
+	FullCrontab        string            `json:"fullCrontab,omitempty"`
+	IncrementalCrontab string            `json:"incrementalCrontab,omitempty"`
+}
+
+type CleanPolicy struct {
+	Name           string `json:"name,omitempty"`
+	RecoveryWindow string `json:"recoveryWindow,omitempty"`
+	Disabled       string `json:"disabled,omitempty"`
+}
+
+type BackupPolicyStatusType string
+
+const (
+	BackupPolicyStatusPreparing BackupPolicyStatusType = "PREPARING"
+	BackupPolicyStatusPrepared  BackupPolicyStatusType = "PREPARED"
+	BackupPolicyStatusRunning   BackupPolicyStatusType = "RUNNING"
+	BackupPolicyStatusFailed    BackupPolicyStatusType = "FAILED"
+	BackupPolicyStatusPaused    BackupPolicyStatusType = "PAUSED"
+	BackupPolicyStatusStopped   BackupPolicyStatusType = "STOPPED"
+)
+
+type BackupDestination struct {
+	Type BackupDestType `json:"type,omitempty"`
+	Path string         `json:"path,omitempty"`
+}
+
+type BackupDestType string
+
+const (
```

Comment:
maybe it's better to define const values in another package

Reply:
The status enum constants will be only used in the recourse and its controllers. I think keeping the definitions in the same file as references would be more clear. Let's discuss :)